nvflare/edge/device/android/app/src/main/java/com/nvidia/nvflare/app/controllers/FlareRunnerController.kt

```diff
@@ -105,66 +117,75 @@ class FlareRunnerController(
-                        Log.e(TAG, "FlareRunnerController: CIFAR-10 data not found in app bundle")
-                        throw TrainingError.DATASET_CREATION_FAILED
-                    } catch (e: DatasetError.InvalidDataFormat) {
-                        Log.e(TAG, "FlareRunnerController: CIFAR-10 data format is invalid")
-                        throw TrainingError.DATASET_CREATION_FAILED
-                    } catch (e: DatasetError.EmptyDataset) {
-                        Log.e(TAG, "FlareRunnerController: CIFAR-10 dataset is empty")
-                        throw TrainingError.DATASET_CREATION_FAILED
-                    } catch (e: Exception) {
-                        Log.e(TAG, "FlareRunnerController: Failed to create CIFAR-10 dataset: $e")
-                        throw TrainingError.DATASET_CREATION_FAILED
+                        Log.d(TAG, "FlareRunnerController: Created XOR dataset")
+                        Log.d(TAG, "FlareRunnerController: XOR dataset size: ${dataset.size()}")
                     }
-                } else if (supportedJobs.contains(SupportedJob.XOR)) {
-                    // Use job name pattern (iOS style) instead of hardcoded "train" phase
-                    jobName = "xor_et"
-                    dataset = dataSource.getDataset(jobName, FlareContext())
-                    Log.d(TAG, "FlareRunnerController: Created XOR dataset")
-                    Log.d(TAG, "FlareRunnerController: XOR dataset size: ${dataset.size()}")
-                } else {
-                    throw TrainingError.DATASET_CREATION_FAILED
                 }
                 
                 // Store the dataset to keep it alive during training
                 currentDataset = dataset
                 Log.d(TAG, "FlareRunnerController: Stored dataset reference: $dataset")
                 
-                // Create FlareRunner with dataset
+                                // Create FlareRunner with dataset
                 val runner = AndroidFlareRunner(
                     context = context,
-                    connection = connection,
+                    connection = createConnection(),
                     jobName = jobName,
                     dataSource = dataSource,
                     deviceInfo = mapOf(
                         "device_id" to (android.provider.Settings.Secure.getString(
-                            context.contentResolver, 
+                            context.contentResolver,
                             android.provider.Settings.Secure.ANDROID_ID
                         ) ?: "unknown"),
                         "platform" to "android",
                         "app_version" to context.packageManager.getPackageInfo(context.packageName, 0).versionName
                     ),
                     userInfo = emptyMap(),
-                    jobTimeout = 86400.0f  // 24 hours
+                    jobTimeout = Float.MAX_VALUE  // No timeout - continue until manual stop
```

Comment:
you can make this configurable if we want user to set, but this is also fine as this is reference implementation of app